Cannot run JMH via gradle from command line

When attempting a `clean jmh` run via gradle it throws an error about not being able to touch the report file.

In the gradle task for jmh we already have an attempt to make sure the directory is there but perhaps this isn't working.

```groovy
    def resultFile = file("build/reports/jmh/result.${format}")
    resultFile.parentFile.mkdirs()
```

After an attempted run there is no `reports` directory under build.

```
$ ./gradlew clean jmh

> Task :jmh FAILED
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.openjdk.jmh.util.Utils (file:/home/nick/.gradle/caches/modules-2/files-2.1/org.openjdk.jmh/jmh-core/1.26/3a60bc5f3b6204cc52139e4cb8967f4209e115e3/jmh-core-1.26.jar) to field java.io.PrintStream.charOut
WARNING: Please consider reporting this to the maintainers of org.openjdk.jmh.util.Utils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
ERROR: org.openjdk.jmh.runner.RunnerException: Can not touch the result file: /home/nick/git-projects/disruptor/build/reports/jmh/result.json
        at org.openjdk.jmh.runner.Runner.internalRun(Runner.java:263)
        at org.openjdk.jmh.runner.Runner.run(Runner.java:210)
        at org.openjdk.jmh.Main.main(Main.java:71)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jmh'.
> Process 'command '/usr/lib/jvm/java-11-openjdk/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
4 actionable tasks: 4 executed
```

--- a/gradle/jmh.gradle
+++ b/gradle/jmh.gradle
@@ -21,7 +21,9 @@ task jmh(type: JavaExec, description: 'Executing JMH benchmarks', group: 'perfor
     def profilers = ['pauses']
     def format = 'json'
     def resultFile = file("build/reports/jmh/result.${format}")
-    resultFile.parentFile.mkdirs()
+    doFirst {
+        resultFile.parentFile.mkdirs()
+    }
 
     args '-rf', format
     args '-rff', resultFile

